platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+                    LOGGER.error("", e);
+                    return getErrorResponse(relayState, authnRequest, StatusCode.AUTHN_FAILED_URI);
+                } catch (WSSecurityException e) {
+                    LOGGER.error("", e);
+                    return getErrorResponse(relayState, authnRequest,
+                            StatusCode.REQUEST_DENIED_URI);
+                } catch (SimpleSign.SignatureException e) {
+                    LOGGER.error("", e);
+                    return getErrorResponse(relayState, authnRequest,
+                            StatusCode.REQUEST_UNSUPPORTED_URI);
+                }
+                LOGGER.debug("Returning Passive & PKI SAML Response.");
+                return getSamlpResponse(relayState, authnRequest, samlpResponse);
+            } else {
+                LOGGER.debug("Building the JSON map to embed in the index.html page for login.");
+                Document doc = DOMUtils.createDocument();
+                doc.appendChild(doc.createElement("root"));
+                String authn = DOM2Writer
+                        .nodeToString(OpenSAMLUtil.toDom(authnRequest, doc, false));
+                String encodedAuthn = RestSecurity.deflateAndBase64Encode(authn);
+                responseMap.put("pki", hasCerts);
+                responseMap.put(SAML_REQ, encodedAuthn);
+                responseMap.put(RELAY_STATE, relayState);
+                String assertionConsumerServiceURL = getAssertionConsumerServiceURL(authnRequest);
+                responseMap.put(ACS_URL, assertionConsumerServiceURL);
+                responseMap.put(SSOConstants.SIG_ALG, signatureAlgorithm);
+                responseMap.put(SSOConstants.SIGNATURE, signature);
+            }
+
+            String json = Boon.toJson(responseMap);
+
+            LOGGER.debug("Returning index.html page.");
+            responseStr = indexHtml.replace(IDP_STATE_OBJ, json);
+            return Response.ok(responseStr).build();
+        } catch (IllegalArgumentException e) {
+            LOGGER.error("", e);
+            if (authnRequest != null) {
+                try {
+                    return getErrorResponse(relayState, authnRequest,
+                            StatusCode.REQUEST_UNSUPPORTED_URI);
+                } catch (IOException | SimpleSign.SignatureException e1) {
+                    LOGGER.error("", e1);
+                }
+            }
+        } catch (UnsupportedOperationException e) {
+            LOGGER.error("", e);
+            if (authnRequest != null) {
+                try {
+                    return getErrorResponse(relayState, authnRequest,
+                            StatusCode.UNSUPPORTED_BINDING_URI);
+                } catch (IOException | SimpleSign.SignatureException e1) {
+                    LOGGER.error("", e1);
+                }
+            }
+        } catch (SimpleSign.SignatureException e) {
+            LOGGER.error("Unable to validate AuthRequest Signature", e);
```

Comment:
Seems inconsistent to have logging messages here and below, but then to not have them in IllegalArgumentException and UnsupportedOperationException catch blocks...although you are returning status codes.